guides/source/sign_up_and_settings.md

```diff
@@ -0,0 +1,1658 @@
+
+Editing Passwords
+-----------------
+
+Now that users can login, let's create all the usual places that users would expect to update their profile, password, email address, and other settings.
+
+### Using Namespaces
+
+The Rails authentication generator already created at `app/controllers/passwords_controller.rb` is the controller for password resets. This means we need to use a different controller for editing passwords of authenticated users.
+
+To prevent conflicts, we can use a feature called **namespaces**. A namespace organizes routes, controllers, and views into folders and helps prevent conflicts like our two passwords controllers.
+
+We'll create a namespace called "Settings" to separate out the user and store settings from the rest of our application.
+
+In `config/routes.rb` we can add the Settings namespace along with a resource for editing passwords:
+
+```ruby
+namespace :settings do
+  resource :password, only: [:show, :update]
+end
+```
+
+This will generate a route for `/settings/password` for editing the current user's password which is separate from the password resets routes at `/password`.
+
+### Adding the Namespaced Passwords Controller & View
+
+Namespaces also move controllers into a matching module in Ruby. This controller will be in a `settings` folder to match the namespace.
+
+Let's create the folder and controller at `app/controllers/settings/passwords_controller.rb` and start with the `show` action.
+
+```ruby
+class Settings::PasswordsController < ApplicationController
+  def show
+  end
+end
+```
+
+Views also move to a `settings` folder so let's create the folder and view at `app/views/settings/passwords/show.html.erb` for this action.
+
+```erb
+<h1>Password</h1>
+
+<%= form_with model: Current.user, url: settings_password_path do |form| %>
+  <% if form.object.errors.any? %>
+    <div><%= form.object.errors.full_messages.first %></div>
+  <% end %>
+
+  <div>
+    <%= form.label :password_challenge %>
+    <%= form.password_field :password_challenge, required: true, autocomplete: "current-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password %>
+    <%= form.password_field :password, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.label :password_confirmation %>
+    <%= form.password_field :password_confirmation, required: true, autocomplete: "new-password" %>
+  </div>
+
+  <div>
+    <%= form.submit "Update password" %>
+  </div>
+<% end %>
+```
+
+We've set the `url:` argument to ensure the form submits to our namespaced route and is processed by the `Settings::PasswordsController`. Passing `model: Current.user` also tells `form_with` to submit a `PATCH` request to process the form with the `update` action.
+
+### Safely Updating Passwords
+
+Let's add that `update` action to the controller now.
+
+```ruby#5-17
+class Settings::PasswordsController < Settings::BaseController
```

Comment:
This is using Settings::BaseController before we introduced it

Reply:
```suggestion
class Settings::PasswordsController < ApplicationController
```